core/src/main/java/org/apache/iceberg/SnapshotParser.java

```diff
@@ -97,19 +105,30 @@ static void toJson(Snapshot snapshot, JsonGenerator generator)
     generator.writeEndObject();
   }
 
+  @Deprecated
   public static String toJson(Snapshot snapshot) {
     // Use true as default value of pretty for backwards compatibility
-    return toJson(snapshot, true);
+    return toJson(snapshot, null,true);
   }
 
+  @Deprecated  // Possibly mark as VisibleForTesting or something.
   public static String toJson(Snapshot snapshot, boolean pretty) {
+    return toJson(snapshot, null, pretty);
+  }
+
+  public static String toJson(Snapshot snapshot, FileIO fileIO) {
+    // Use true as default value of pretty for backwards compatibility
+    return toJson(snapshot, fileIO, true);
+  }
+
+  public static String toJson(Snapshot snapshot, FileIO fileIO, boolean pretty) {
```

Comment:
I think the suggested changes in BaseSnapshot avoid needing to add this here.

Reply:
Reverting entire file to what's in master and removed the deprecation.